projects/ad9361/src/ad9361_tinyiiod.c

```diff
@@ -0,0 +1,2721 @@
+				bytes_no += sprintf(buf + bytes_no, " ");
+			}
+			bytes_no += sprintf(buf + bytes_no, "%s", ad9361_rf_rx_port[i]);
+			if(bytes_no < 0) {
+				break;
+			}
+		}
+		return bytes_no;
+	}
+}
+
+/**
+ * get_rf_port_select
+ * @param *buff where value is stored
+ * @param len maximum length of value to be stored in buf
+ * @param *channel channel properties
+ * @return length of chars written in buf, or negative value on failure
+ */
+ssize_t get_rf_port_select(char *buf, size_t len,
+			   const struct channel_info *channel)
+{
+	ssize_t ret = 0;
+	if(channel->ch_out) {
+		uint32_t mode;
+		ret = ad9361_get_tx_rf_port_output(ad9361_phy, &mode);
+		return ret < 0 ? ret : sprintf(buf, "%s", ad9361_rf_tx_port[mode]);
+	} else {
+		uint32_t mode;
+		ret = ad9361_get_rx_rf_port_input(ad9361_phy, &mode);
+		return ret < 0 ? ret : sprintf(buf, "%s", ad9361_rf_rx_port[mode]);
+	}
+}
+
+/**
+ * get_gain_control_mode_available
+ * @param *buff where value is stored
+ * @param len maximum length of value to be stored in buf
+ * @param *channel channel properties
+ * @return length of chars written in buf, or negative value on failure
+ */
+ssize_t get_gain_control_mode_available(char *buf, size_t len,
+					const struct channel_info *channel)
+{
+	return (ssize_t) sprintf(buf, "%s %s %s %s",
+				 ad9361_agc_modes[0],
+				 ad9361_agc_modes[1],
+				 ad9361_agc_modes[2],
+				 ad9361_agc_modes[3]);
+}
+
+/**
+ * get_gain_control_mode
+ * @param *buff where value is stored
+ * @param len maximum length of value to be stored in buf
+ * @param *channel channel properties
+ * @return length of chars written in buf, or negative value on failure
+ */
+ssize_t get_gain_control_mode(char *buf, size_t len,
+			      const struct channel_info *channel)
+{
+	return (ssize_t) sprintf(buf, "%s",
+				 ad9361_agc_modes[ad9361_phy->agc_mode[channel->ch_num]]);
+}
+
+/**
+ * get_hardwaregain_available
+ * @param *buff where value is stored
+ * @param len maximum length of value to be stored in buf
+ * @param *channel channel properties
+ * @return length of chars written in buf, or negative value on failure
+ */
+ssize_t get_hardwaregain_available(char *buf, size_t len,
+				   const struct channel_info *channel)
+{
+	if (channel->ch_out) {
+		return (ssize_t) snprintf(buf, len, "[%d, %d, %d]", 0, 250, 89750);
+	} else {
+		return (ssize_t) snprintf(buf, len, "[%"PRIi32", %d, %"PRIi32"]",
+					  ad9361_phy->rx_gain[ad9361_phy->current_table].starting_gain_db,
+					  1,
+					  ad9361_phy->rx_gain[ad9361_phy->current_table].max_gain_db);
+	}
+}
+
+/**
+ * get_hardwaregain
+ * @param *buff where value is stored
+ * @param len maximum length of value to be stored in buf
+ * @param *channel channel properties
+ * @return length of chars written in buf, or negative value on failure
+ */
+ssize_t get_hardwaregain(char *buf, size_t len,
+			 const struct channel_info *channel)
+{
+	if(channel->ch_out) {
+		int32_t ret = ad9361_get_tx_atten(ad9361_phy, channel->ch_num + 1);
```

Comment:
If 'ret' is needed in both cases should it just be defined at the beginning of the function?